include/trie.h

```diff
@@ -0,0 +1,107 @@
+/*
+    A trie data structure
+*/
+
+#ifndef INCLUDE_TRIE_H_
+#define INCLUDE_TRIE_H_
+
+#include <stdio.h>
+#include <string.h>
+#include <stdlib.h>
+#include <assert.h>
+#include "parser.h"
+
+typedef struct trie_t trie_t;
+struct trie_t {
+    char current; 
+        // The first trie_t will be '/0' for any Trie.
+    trie_t **children;
+         // ALPHABET_SIZE is 256 for all possible characters.
+    int is_word; 
+        // if is_word is 1, indicates that this is the end of a word. Otherwise 0.
+    trie_t *parent;
+        // parent trie_t for traversing backwards
+};
+
+/*
+    Creates and allocates memory for new trie_t.
+    
+    Parameters:
+     - current: A char for the current character
+    
+    Returns:
+     - A pointer to the trie, or NULL if a pointer 
+       cannot be allocated
+*/
+trie_t *new_trie (char current);
```

Comment:
This function should be named `trie_new`.

Reply:
fixed